cfme/services/catalogs/catalog_item.py

```diff
@@ -112,23 +197,11 @@ def __init__(self, item_type=None, vm_name=None, name=None, description=None,
         self.provider_type = provider_type
         self.provisioning_data = prov_data
         self.domain = domain
-        Navigatable.__init__(self, appliance=appliance)
-
-    def __str__(self):
-        return self.name
 
     def create(self):
-        # Create has sequential forms, the first is only the provider type
-        navigate_to(self, 'Add')
-        # For element not found exception (To be removed)
-        sel.sleep(5)
+        view = navigate_to(self, 'Add')
         sel.select("//select[@id='st_prov_type']",
                    self.provider_type or self.item_type or 'Generic')
```

Comment:
Check out the before_fill method for the views, I think this should move there. I used it for the cloud instance provisioning form here:
https://github.com/ManageIQ/integration_tests/pull/4013/files#diff-38f02157b3579c25d87ef2e056f96b6fR386